pyccel/ast/core.py

```diff
@@ -197,6 +197,18 @@ def _eval_subs(self, old, new):
     def _eval_evalf(self,prec):
         return sp_Pow(self.base,self.exp).evalf(prec)
 
+    @property
+    def is_real(self):
+        return self._args[0].is_real and self._args[1].is_real
```

Comment:
Should this not be or?
2^0.5 is real
1.5^2 is also real

Or does fortran prevent mixing types like that?

Reply:
yes you are right

Reply:
if it's an integer => it's a real

Reply:
Ah ok sorry